app/assets/javascripts/user_analytics.js

```diff
@@ -0,0 +1,185 @@
+//
+// stats
+//
+function statistics() {
+  return JSON.parse(document
+    .getElementById('statistics')
+    .attributes
+    .getNamedItem('data-statistics')
+    .value);
+}
+
+function dailyStatistics() {
+  return Object.entries(statistics()
+    .daily
+    .grouped_by_date);
+}
+
+//
+// dates
+//
+function formatDate(date) {
+  return date.toJSON().slice(0, 10);
+}
+
+function today() {
+  return new Date();
+}
+
+function yesterday() {
+  var todayDate = today();
+  return new Date(todayDate.setDate(todayDate.getDate() - 1));
+}
+
+function formattedTomorrowDate() {
+  return statistics().metadata.formatted_next_date;
```

Comment:
Is this also a string, like `formattedTodayString`? Can we consistent-ize the names if so?

Reply:
Addressed JS things here: https://github.com/simpledotorg/simple-server/pull/843/commits/ad4d217691c22b825a7a76e43af3099e9be2fa03. This includes moving to ES6 best practices (no usage of `var`s anywhere).